2.0.0-beta.22 breaks deploy to GitHub Pages (shallow update not allowed)

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Updating to docusaurus v2.0.0-beta.22 breaks my deployment to GitHub Pages.

The relevant commandline error follows:
```
[INFO] `git commit -m "Deploy website - based on 6f67bb9205e05e2d37ba6368e30774c48e6db78d"` code: 0
To github.com:yechs/website.git
 ! [remote rejected] master -> master (shallow update not allowed)
error: failed to push some refs to 'github.com:yechs/website.git'
[INFO] `git push --force origin master` code: 1
[ERROR] Deployment of the build output failed.
[ERROR] Error: Running "git push" command failed. Does the GitHub user account you are using have push access to the repository?
    at runDeploy (/Users/yeshu/Coding/Personal/website/node_modules/@docusaurus/core/lib/commands/deploy.js:168:19)
    at async Command.deploy (/Users/yeshu/Coding/Personal/website/node_modules/@docusaurus/core/lib/commands/deploy.js:189:13)
error Command failed with exit code 1.
```

According to the command line output, it looks like a shallow update problem. However, the weird thing is, docusaurus has been doing shallow clone deployments for almost a year now and has never failed before.
Relevant PR: https://github.com/facebook/docusaurus/pull/5748

### Logs

If you want a full log, there is one available from GitHub Actions (https://github.com/yechs/website/runs/7256462347)

And here is the successful log running `yarn deploy` on v2.0.0-beta.21 three days ago (https://github.com/yechs/website/runs/7204294330)

### Reproduction

Just to rule out the variables of GitHub Actions or GitHub changing their behavior:

I've ran `yarn deploy` locally on v2.0.0-beta.22 and it failed with the same message.
I have also manually reverted back to v2.0.0-beta.21 and ran `yarn deploy`, which functioned perfectly normal. (Note that it is doing the same shallow clone)

### Reproducible demo

https://github.com/yechs/website/

### Steps to reproduce

1. Run `yarn deploy`
2. Observe that it fails
3. `git checkout 193d6e0` to go back to docusaurus v2.0.0-beta.21
4. Run `yarn` to install dependencies
5. Run `yarn deploy`
6. Observe that it succeeds

### Expected behavior

Deployment continues to function as in v2.0.0-beta.21

### Actual behavior

Deployment fails. See log from [GitHub Actions](https://github.com/yechs/website/runs/7256462347).

### Your environment

- Public source code:
  - Source code at https://github.com/yechs/website/
  - Deployed to https://github.com/yechs/yechs.github.io/
- Public site URL:
- Docusaurus version used:
  - v2.0.0-beta.21
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):
  - Ubuntu (GitHub Actions) with Node v16
  - MacOS with Node v16.15.1
  - Arch Linux with Node v16.14.2


### Self-service

- [ ] I'd be willing to fix this bug myself.

--- a/packages/docusaurus/src/commands/deploy.ts
+++ b/packages/docusaurus/src/commands/deploy.ts
@@ -183,9 +183,6 @@ You can also set the deploymentBranch property in docusaurus.config.js .`);
   // Save the commit hash that triggers publish-gh-pages before checking
   // out to deployment branch.
   const currentCommit = shellExecLog('git rev-parse HEAD').stdout.trim();
-  const currentGitRootDir = shellExecLog(
-    'git rev-parse --show-toplevel',
-  ).stdout.trim();
 
   const runDeploy = async (outputDirectory: string) => {
     const fromPath = outputDirectory;
@@ -216,16 +213,6 @@ You can also set the deploymentBranch property in docusaurus.config.js .`);
       logger.error`Copying build assets from path=${fromPath} to path=${toPath} failed.`;
       throw err;
     }
-
-    const gitConfigFromPath = path.join(currentGitRootDir, `.git`, `config`);
-    const gitConfigToPath = path.join(toPath, `.git`, `config`);
-    try {
-      await fs.copy(gitConfigFromPath, gitConfigToPath);
-    } catch (err) {
-      logger.error`Copying git config from path=${gitConfigFromPath} to path=${gitConfigToPath} failed.`;
-      throw err;
-    }
-
     shellExecLog('git add --all');
 
     const commitMessage =

